pkg/token/provider.go

```diff
@@ -15,7 +15,7 @@ type TokenProvider interface {
 	Token() (adal.Token, error)
 }
 
-func newTokenProvider(o *Options) (TokenProvider, error) {
+func NewTokenProvider(o *Options) (TokenProvider, error) {
```

Comment:
So instead of exposing the GetToken from `pkg` directly, how about we convert to a structure like this:

1. rename `pkg` to `internal`, this can avoid library usage to import the internal implementation. 
2. export a minimum `GetTokenOptions` struct at the package root level.
3. export the `GetToken` function at the package root level, and let client use like this: `kubelogin.GetToken(kubelogin.GetTokenOptions{})`. In the `GetToken`, we convert the exported `GetTokenOptions` to the internal options object and invoke the internal `GetToken` call.
4. future library usage goes to package root level, cli usage stays the same.

kubelogin was not designed to be consumed as library at this first place, so I think it would be better to hide most of the things first so we can avoid unexpected dependencies.

@weinong @turkenh @erhancagirici WDYT?

Reply:
hey all, I created a proposal for above changes: https://github.com/Azure/kubelogin/issues/373 @turkenh  / @erhancagirici Can you take a look and let us know your thoughts? If no blocker, we will start the implementations and migrations, thanks!